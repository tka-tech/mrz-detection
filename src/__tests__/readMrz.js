'use strict';

const path = require('path');

const Image = require('image-js').Image;

const { getMrz, readMrz } = require('../..');

it(
  'test the extraction of MRZ characters on an identity card',
  async () => {
    const img = await Image.load(path.join(__dirname, 'fixtures/id1.jpg'));
    let mrzImage = getMrz(img);
    // mrzImage.save('./test.jpg');
    let mrz = await readMrz(mrzImage);
    expect(mrz).toMatchSnapshot();
  },
  15000
);

it(
  'test the extraction of MRZ characters on an identity card 2',
  async () => {
    const img = await Image.load(path.join(__dirname, 'fixtures/id2.jpg'));
    let mrzImage = getMrz(img);
    let mrz = await readMrz(mrzImage);
    expect(mrz).toMatchSnapshot();
  },
  15000
);
