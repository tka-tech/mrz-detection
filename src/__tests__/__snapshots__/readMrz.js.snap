// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test the extraction of MRZ characters on an identity card 1`] = `
Array [
  "IDFRABERTHIER<<<<<<<<<<<<<<<<<<<<<<<",
  "9409923102854CORINNE<<<<<<<6512068F4",
]
`;

exports[`test the extraction of MRZ characters on an identity card 2 1`] = `
Array [
  "P<ESTMAENNIK<<MARI<LIIS<<<<<<<<<<<<<<<<<<<<<",
  "KB00000003EST7501229F120522647501220010<<<02",
]
`;
